core/lib/slider.ts

```diff
@@ -0,0 +1,569 @@
+								value[handleIndex] = max$();
+								return value;
+							});
+							break;
+						case 'PageUp':
+							// TODO it is optional in accessibility guidelines, so define the skip value for steps and write unit test
+							break;
+						case 'PageDown':
+							// TODO it is optional in accessibility guidelines, so define the skip value for steps and write unit test
+							break;
+						default:
+							return;
+					}
+					event.preventDefault();
+					event.stopPropagation();
+				}
+			},
+		},
+	};
+}
+
+export interface SliderHandleCommonPropsAndState extends WidgetsCommonPropsAndState {
+	/**
+	 * If `true` the hanlde needs to be horizontally aligned, otherwise vertically
+	 */
+	vertical: boolean;
+
+	/**
+	 * Left offset of the handle in %
+	 */
+	positionLeft: number[];
+
+	/**
+	 * Top offset of the handle in %
+	 */
+	positionTop: number[];
+
+	/**
+	 * The id of the slider handle.
+	 */
+	handleId: number;
+
+	/**
+	 * If `true` handle is focusable but not actionable
+	 */
+	readonly: boolean;
+
+	/**
+	 * If `true` handle is neither focusable nor actionable
+	 */
+	disabled: boolean;
+}
+
+export interface SliderHandleProps extends SliderHandleCommonPropsAndState {
+	/**
+	 * MouseDown event handler
+	 * @param event - mouse event
+	 */
+	onMouseDown(event: MouseEvent): void;
+
+	/**
+	 * Keyboard event handler
+	 * @param event - keyboard event
+	 */
+	onKeyDown(event: KeyboardEvent, handleId: number): void;
+}
+
+export interface SliderHandleState extends SliderHandleCommonPropsAndState {
+	/**
+	 * Needed to avoid changedetection on the mouse move over the same coordinate (vertical or horizonal movement)
+	 */
+	prevCoordinate: number;
+}
+
+export interface SliderHandleActions {}
+
+export interface SliderHandleApi {}
+
+export interface SliderHandleDirectives {
+	/**
+	 * Directive to be put on the slider handle. It adds the handle to the slider.
+	 */
+	sliderHandleDirective: Directive;
+}
+
+export type SliderHandleWidget = Widget<SliderHandleProps, SliderHandleState, SliderHandleApi, SliderHandleActions, SliderHandleDirectives>;
+
+const defaultHandleConfig: SliderHandleProps = {
+	handleId: 0,
+	className: '',
+	vertical: defaultSliderConfig.vertical,
+	readonly: defaultSliderConfig.readonly,
+	disabled: defaultSliderConfig.disabled,
+	positionLeft: [0],
+	positionTop: [0],
+	onMouseDown: noop,
+	onKeyDown: noop,
+};
+
+const configSliderHandleValidator: ConfigValidator<SliderHandleProps> = {
+	className: typeString,
+	vertical: typeBoolean,
+	positionLeft: typeArray,
+	positionTop: typeArray,
+	onMouseDown: typeFunction,
+	onKeyDown: typeFunction,
+};
+
+const sliderHandleProps = Object.keys(defaultHandleConfig) as (keyof SliderHandleProps)[];
+
+function createSliderHandle(config?: PropsConfig<SliderHandleProps>): SliderHandleWidget {
+	const [{onMouseDown$, onKeyDown$, ...stateProps}, patch] = writablesForProps(defaultHandleConfig, config, configSliderHandleValidator);
+	const prevCoordinate$ = writable(-1);
```

Comment:
Is this used anywhere?

Reply:
Also the whole prevCoordinate handling and it being a part of the state looks... hmm... fishy :)

Reply:
oh, yes, forgot to remove it from the state (it is only used in the angular handle implementation, so should not be here)